Library/Homebrew/utils/shell.rb

```diff
@@ -158,7 +158,20 @@ def shell_with_prompt(type, preferred_path:, notice:)
       preferred = from_path(preferred_path)
       subshell = case preferred
       when :zsh
-        "PROMPT='%B%F{green}#{type}%f %F{blue}$%f%b ' RPROMPT='[%B%F{red}%~%f%b]' #{preferred_path} -f"
+        zdotdir = Pathname.new(HOMEBREW_TEMP/"brew-zsh-prompt")
+        zdotdir.mkpath
+        FileUtils.chmod_R(0700, zdotdir)
+        FileUtils.cp(HOMEBREW_LIBRARY_PATH/"utils/zsh/brew-sh-prompt-zshrc.zsh", zdotdir/".zshrc")
```

Comment:
Might be safer to at least bring `#{Process.euid}` into the pathname or similar so we make sure the path is per-user.

Reply:
@Bo98 Is `HOMEBREW_TEMP` not the per-user temp? If not: ideally we'd do that and: yes, agreed.

Reply:
> Is `HOMEBREW_TEMP` not the per-user temp?

In Homebrew, no it's global. We may switch to being per-user here on macOS, but it'll still be global on Linux.